package/MDAnalysis/lib/formats/libdcd.pyx

```diff
@@ -0,0 +1,414 @@
+
+cdef enum:
+    FIO_READ = 0x01
+    FIO_WRITE = 0x02
+
+cdef enum:
+    DCD_IS_CHARMM       = 0x01
+    DCD_HAS_4DIMS       = 0x02
+    DCD_HAS_EXTRA_BLOCK = 0x04
+
+DCD_ERRORS = {
+    0: 'No Problem',
+    -1: 'Normal EOF',
+    -2: 'DCD file does not exist',
+    -3: 'Open of DCD file failed',
+    -4: 'read call on DCD file failed',
+    -5: 'premature EOF found in DCD file',
+    -6: 'format of DCD file is wrong',
+    -7: 'output file already exiss',
+    -8: 'malloc failed'
+}
+
+cdef extern from 'include/fastio.h':
+    int fio_open(const char *filename, int mode, fio_fd *fd)
+    int fio_fclose(fio_fd fd)
+    fio_size_t fio_ftell(fio_fd fd)
+    fio_size_t fio_fseek(fio_fd fd, fio_size_t offset, int whence)
+
+cdef extern from 'include/readdcd.h':
+    int read_dcdheader(fio_fd fd, int *n_atoms, int *nsets, int *istart,
+                       int *nsavc, double *delta, int *nfixed, int **freeind,
+                       float **fixedcoords, int *reverse_endian, int *charmm,
+                       char **remarks, int *len_remarks)
+    void close_dcd_read(int *freeind, float *fixedcoords)
+    int read_dcdstep(fio_fd fd, int n_atoms, float *X, float *Y, float *Z,
+                     double *unitcell, int num_fixed,
+                     int first, int *indexes, float *fixedcoords,
+                     int reverse_endian, int charmm)
+    int read_dcdsubset(fio_fd fd, int n_atoms, int lowerb, int upperb,
+                     float *X, float *Y, float *Z,
+                     double *unitcell, int num_fixed,
+                     int first, int *indexes, float *fixedcoords,
+                     int reverse_endian, int charmm)
+    int write_dcdheader(fio_fd fd, const char *remarks, int natoms, 
+                   int istart, int nsavc, double delta, int with_unitcell, 
+                   int charmm);
+    int write_dcdstep(fio_fd fd, int curstep, int curframe, 
+			 int natoms, const float *x, const float *y, const float *z,
+			 const double *unitcell, int charmm);
+
+DCDFrame = namedtuple('DCDFrame', 'x unitcell')
+
+cdef class DCDFile:
+    cdef fio_fd fp
+    cdef readonly fname
+    cdef int is_open
+    cdef readonly int n_atoms
+    cdef int nsets
+    cdef readonly int istart
+    cdef readonly int nsavc
+    cdef readonly double delta
+    cdef readonly int nfixed
+    cdef int *freeind
+    cdef float *fixedcoords
+    cdef int reverse_endian
+    cdef int charmm
+    cdef str mode
+    cdef readonly int n_dims
+    cdef readonly int n_frames
+    cdef bint b_read_header
+    cdef int current_frame
+    cdef readonly remarks
+    cdef int reached_eof
+    cdef readonly int firstframesize
+    cdef readonly int framesize
+    cdef readonly int header_size
+
+    def __cinit__(self, fname, mode='r'):
+        self.fname = fname.encode('utf-8')
+        self.n_atoms = 0
+        self.is_open = False
+        if path.isfile(self.fname) and mode == 'w':
+            raise IOError('''Aborting -- attempted to overwrite an
```

Comment:
I think we can allow to overwrite files. This is what we do with other files as well and python file objects don't raise an issue or warn either.